### PR TITLE
[vcpkg docs] Fix the vcpkg_install_msbuild docs to reflect handling of x86==Win32.

### PR DESCRIPTION
**Pull request by BillyONeal**
Also fixes ports that were doing this in an overcomplicated way.

See:
* https://github.com/microsoft/vcpkg/pull/28119#issuecomment-1336270264
* https://github.com/microsoft/vcpkg/issues/28076

### CODE REVIEW

File: ports/activemq-cpp/portfile.cmake [hunk 1]
```diff
@@ -65,12 +65,10 @@ else()
         vcpkg_replace_string("${ACTIVEMQCPP_MSVC_PROJ}" ";libapr-1.lib" ";apr-1.lib")
     endif()
 
-    if (VCPKG_TARGET_ARCHITECTURE MATCHES "x86")
+    if(VCPKG_TARGET_ARCHITECTURE MATCHES "x86")
```
Comment (LilyWangLL) [inline ports/activemq-cpp/portfile.cmake:68]: Should this judgment `x86` also be removed?

File: scripts/templates/portfile.in.cmake [hunk 2]
```diff
@@ -13,7 +13,6 @@
 #   VCPKG_TOOLCHAIN           = ON OFF
 #   TRIPLET_SYSTEM_ARCH       = arm x86 x64
 #   BUILD_ARCH                = "Win32" "x64" "ARM"
```
Comment (LilyWangLL) [inline scripts/templates/portfile.in.cmake:15]: Why is `arm64` not added here? Is it included in `ARM`?

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: scripts/templates/portfile.in.cmake (hunk 2 above)
Comment (LilyWangLL) [inline scripts/templates/portfile.in.cmake:15]: Why is `arm64` not added here? Is it included in `ARM`?

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/ecsutil/portfile.cmake [hunk 3]
```diff
@@ -24,22 +20,22 @@ vcpkg_from_github(
 )
 
 vcpkg_install_msbuild(
-    SOURCE_PATH ${SOURCE_PATH}
+    SOURCE_PATH "${SOURCE_PATH}"
     PROJECT_SUBPATH ECSUtil.sln
-    PLATFORM ${PLATFORM}
+    PLATFORM "${TRIPLET_SYSTEM_ARCH}"
```
Comment (LilyWangLL) [inline ports/ecsutil/portfile.cmake:25]:
```suggestion
```
This is a default behavior, why not remove this like ports `3fd` and `ace`?

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by ras0219-msft** [comment]
Docs changes have been migrated in https://github.com/microsoft/vcpkg-docs/pull/21

**Comment by JonLiu1993** [comment]
@BillyONeal, The modification of the documentation part has been merged in microsoft/vcpkg-docs#21, can we remove the modification related to the documentation in this PR?

**Comment by BillyONeal** [comment]
Was merged

**Pull request closed without merging**